scanner/scanner.go

```diff
@@ -108,6 +117,101 @@ func (s *Scanner) ScanAndClean(opts ScanOptions) (*Context, error) {
 	return c, nil
 }
 
+func (s *Scanner) ExecuteWatch() error {
+	var err error
+	s.watcher, err = fsnotify.NewWatcher()
+	if err != nil {
+		log.Fatalf("error creating watcher: %v\n", err)
+	}
```

Comment:
does it make sense to return an error here instead of quiting? that way main() can continue unrolling its defer stack for cleanups